nat-lab/tests/telio.py

```diff
@@ -797,6 +798,27 @@ async def set_mesh_off(self):
     async def receive_ping(self) -> str:
         return await asyncio.to_thread(self.get_proxy().receive_ping)
 
+    async def wait_for_listen_port_ready(
+        self,
+        protocol: str,
+        port: int,
+        process: str = "python3",
+        timeout: Optional[float] = None,
+    ) -> None:
+        assert (
+            self._connection.target_os == TargetOS.Linux
+        ), "Waiting for listen ports is supported only on Linux hosts"
+
+        if not await CommandGrepper(
+            self._connection,
+            [
+                "netstat",
+                "-lpnu",
```

Comment:
We provide a "protocol" argument but the netstat command passes "-u" flag, listing only udp sockets

Reply:
or you're differentiating "udp" vs. "udp6"? 🤔

Reply:
nice catch!

I was thinking about `udp` for this speciic case, but wanted to make the function general. Hence the mistake. I will fix it.

Reply:
Also I just realized that port `:123` would match anything `:123x` or `:123xx` as well.

Reply:
pushed